File: 3522-find-the-power-of-k-size-subarrays-i/3522-find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int> power;
        int l = 0;
        int r = k-1;
        while(r<n){    
            bool isSorted = true;
            int maxi = -1;
            for(int i=l; i<r; i++){
                if(nums[i]+1!=nums[i+1]){
                    isSorted = false;
                    break;
                }
            }
            if(isSorted){
                maxi = nums[r];
            }
            power.push_back(maxi);
            l++;
            r++;
        }

        return power;
    }
};

File: 3522-find-the-power-of-k-size-subarrays-i/3522-find-the-power-of-k-size-subarrays-i.py
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        l=0
        r = k-1
        res = []
        while r < n :
            is_sorted = True
            maxi = -1
            for i in range(l,r):
                if nums[i]+1!=nums[i+1]:
                    is_sorted = False
                    break
            
            if is_sorted:
                maxi = nums[r]
            res.append(maxi)
            r+=1
            l+=1
        return res